src/vhbb.cpp

```diff
@@ -55,6 +55,8 @@ void network_test() {
 
 int main() {
   sceIoMkdir(VHBB_DATA.c_str(), 0777);
+  auto sce_common_dialog_config_param = SceCommonDialogConfigParam{};
+  sceCommonDialogSetConfigParam(&sce_common_dialog_config_param);
```

Comment:
This is ok for the time being. But since only one dialog can be shown at a time, IME should be made into a singleton view and this code should then be moved into the constructor.

Reply:
If you make it a singleton, how to handle the result pointer passed to the view? A separate setter that has to be called before it's added to the activity? And how to handle the "want to be destroyed" flag with a singleton?

Reply:
The method starting the dialog would return a [`std::future`](http://www.cplusplus.com/reference/future/promise/).

The caller could then fetch that result once it's available.

Reply:
Cannot use future without pretty complicated threading logic because it's blocking.